clients/common/cblas_interface.cpp

```diff
@@ -109,6 +110,32 @@ extern "C" {
 
     //axpy
     template<>
+    void cblas_axpy<rocblas_half>( rocblas_int n,
+                            rocblas_half alpha,
+                            rocblas_half *x, rocblas_int incx,
+                            rocblas_half *y, rocblas_int incy)
+    {
+        rocblas_int abs_incx = incx >=0 ? incx : -incx;
+        rocblas_int abs_incy = incy >=0 ? incy : -incy;
+        float alpha_float = static_cast<float>(alpha);
+        std::unique_ptr<float []> x_float(new float[n*abs_incx]());
```

Comment:
I think its worth the effort to pack the temp buffers; you can create terrible performance edge cases with sparse vectors.  The increase in complexity would be worthwhile to me.